control/control.go

```diff
@@ -1016,6 +1016,54 @@ func (p *pluginControl) CollectMetrics(id string, allTags map[string]map[string]
 	return
 }
 
+func (p *pluginControl) StreamMetrics(
+	id string,
+	allTags map[string]map[string]string,
+	maxCollectDuration time.Duration,
+	maxMetricsBuffer int64) (chan []core.Metric, chan error, []error) {
+	if !p.Started {
+		return nil, nil, []error{ErrControllerNotStarted}
+	}
+	errs := make([]error, 0)
+	pluginToMetricMap, serrs, err := p.subscriptionGroups.Get(id)
+	if err != nil {
+		controlLogger.WithFields(log.Fields{
+			"_block":                "StreamMetrics",
+			"subscription-group-id": id,
+		}).Error(err)
+		errs = append(errs, err)
+		return nil, nil, errs
+	}
+
+	if serrs != nil {
+		for _, e := range serrs {
+			errs = append(errs, e)
+		}
+	}
+	if len(pluginToMetricMap) > 1 {
```

Comment:
if you still want to see previous "serrs", you cannot change it. Otherwise "len(pluginToMetricMap) > 1" is a broken error which should be handled earlier. Anyway, it does not hurt much to keep this way.